### ISSUE TITLE
Fails to compile on Linux with "Mutex creation failed" error

### ISSUE THREAD

**Issue by joeroe** [mfasiolo/qgam#28]
Trying to install qgam on Manjaro Linux 17.1.12 fails:

```
> install.packages("qgam")
Installing package into ‘/home/joeroe/R/x86_64-pc-linux-gnu-library/3.5’
(as ‘lib’ is unspecified)
trying URL 'https://cran.rstudio.com/src/contrib/qgam_1.2.2.tar.gz'
Content type 'application/x-gzip' length 927339 bytes (905 KB)
==================================================
downloaded 905 KB

* installing *source* package ‘qgam’ ...
** package ‘qgam’ successfully unpacked and MD5 sums checked
** libs
gcc -I"/usr/include/R/" -DNDEBUG   -D_FORTIFY_SOURCE=2   -fpic  -march=x86-64 -mtune=generic -O2 -pipe -fstack-protector-strong -fno-plt  -c init.c -o init.o
gcc -I"/usr/include/R/" -DNDEBUG   -D_FORTIFY_SOURCE=2   -fpic  -march=x86-64 -mtune=generic -O2 -pipe -fstack-protector-strong -fno-plt  -c mgcv_wrap.c -o mgcv_wrap.o
gcc -shared -L/usr/lib64/R/lib -Wl,-O1,--sort-common,--as-needed,-z,relro,-z,now -o qgam.so init.o mgcv_wrap.o -L/usr/lib64/R/lib -lR
installing to /home/joeroe/R/x86_64-pc-linux-gnu-library/3.5/qgam/libs
** R
** data
** inst
** byte-compile and prepare package for lazy loading
terminate called after throwing an instance of 'std::runtime_error'
  what():  Mutex creation failed
/usr/lib64/R/bin/INSTALL: line 34: 29302 Done                    echo 'tools:::.install_packages()'
     29303 Aborted                 (core dumped) | R_DEFAULT_PACKAGES= LC_COLLATE=C "${R_HOME}/bin/R" $myArgs --slave --args ${args}
Warning in install.packages :
  installation of package ‘qgam’ had non-zero exit status

The downloaded source packages are in
	‘/tmp/RtmpzHREJU/downloaded_packages’
```

I assume this is related to the release of glibc 2.28-1, which also affected the later package (issues [#45](/r-lib/later/issues/45) and [#63](/r-lib/later/issues/63), now fixed). In which case it will also be a problem on Arch Linux and Alpine.

**Comment by mfasiolo** [comment]
Thanks for pointing this out. However at the moment I wouldn't know how to fix this!

**Comment by joeroe** [comment]
Seems to have fixed itself, as of Manjaro 18.0 / Arch 2018.12.01. Thanks for the useful package.
